presto-druid/src/test/java/io/prestosql/plugin/druid/TestingDruidServer.java

```diff
@@ -0,0 +1,286 @@
+            throw new RuntimeException(e);
+        }
+        finally {
+            cleanDirectory(new File(hostWorkingDirectory));
+        }
+    }
+
+    public String getJdbcUrl()
+    {
+        return getJdbcUrl(broker.getMappedPort(DRUID_BROKER_PORT));
+    }
+
+    public int getCoordinatorOverlordPort()
+    {
+        return coordinator.getMappedPort(DRUID_COORDINATOR_PORT);
+    }
+
+    private static String getJdbcUrl(int port)
+    {
+        return format("jdbc:avatica:remote:url=http://localhost:%s/druid/v2/sql/avatica/", port);
+    }
+
+    void ingestData(String datasource, String indexTaskFile, String dataFilePath)
+            throws IOException, InterruptedException
+    {
+        middleManager.withCopyFileToContainer(forHostPath(dataFilePath),
+                getMiddleManagerContainerPathForDataFile(dataFilePath));
+        String indexTask = Resources.toString(getResource(indexTaskFile), Charset.defaultCharset());
+
+        Request.Builder requestBuilder = new Request.Builder();
+        requestBuilder.addHeader("content-type", "application/json;charset=utf-8")
+                .url("http://localhost:" + getCoordinatorOverlordPort() + "/druid/indexer/v1/task")
+                .post(RequestBody.create(null, indexTask));
+        Request ingestionRequest = requestBuilder.build();
+        Response response = null;
+        try {
+            response = httpClient.newCall(ingestionRequest).execute();
+            Assert.assertTrue(checkDatasourceAvailable(datasource), "Datasource " + datasource + " not loaded");
+        }
+        finally {
+            if (response != null) {
+                response.close();
+            }
+        }
+    }
+
+    private boolean checkDatasourceAvailable(String datasource)
+            throws IOException, InterruptedException
+    {
+        Map<String, Double> datasourceAvailabilityDetails = null;
+        boolean datasourceNotLoaded = true;
+        int attempts = 10;
+        while (datasourceNotLoaded && attempts > 0) {
+            Request.Builder requestBuilder = new Request.Builder();
+            requestBuilder.url("http://localhost:" + getCoordinatorOverlordPort() + "/druid/coordinator/v1/loadstatus")
+                    .get();
+            Request datasourceAvailabilityRequest = requestBuilder.build();
+            try (Response response = httpClient.newCall(datasourceAvailabilityRequest).execute()) {
+                ObjectMapper mapper = new ObjectMapper();
+                datasourceAvailabilityDetails = mapper.readValue(response.body().string(), Map.class);
+                datasourceNotLoaded = datasourceAvailabilityDetails.get(datasource) == null || Double.compare(datasourceAvailabilityDetails.get(datasource), 100.0) < 0;
+                if (datasourceNotLoaded) {
+                    attempts--;
+                    // Wait for some time since it can take a while for coordinator to load the ingested segments
+                    Thread.sleep(15000);
```

Comment:
We do the check here:
```
datasourceNotLoaded = datasourceAvailabilityDetails.get(datasource) == null || Double.compare(datasourceAvailabilityDetails.get(datasource), 100.0) < 0;
                if (datasourceNotLoaded) {
                    attempts--;
                    // Wait for some time since it can take a while for coordinator to load the ingested segments
                    Thread.sleep(15000);
                }
```

Reply:
Cool, if we can verify the condition is satisfied, this can benefit in robustness. Then you can also more frequently for faster test startup.

Reply:
Not sure I understand the concern here. We check whether datasource is available by using the druid coordinator api. If it isn't available, we sleep for 15 seconds before checking again. In total, we wait for up to 150 seconds before bailing out. 
```
int attempts = 10;
while (datasourceNotLoaded && attempts > 0) {
....
        ObjectMapper mapper = new ObjectMapper();
        datasourceAvailabilityDetails = mapper.readValue(response.body().string(), Map.class);
         
        datasourceNotLoaded = datasourceAvailabilityDetails.get(datasource) == null || 
             Double.compare(datasourceAvailabilityDetails.get(datasource), 100.0) < 0;
        if (datasourceNotLoaded) {
              attempts--;
              // Wait for some time since it can take a while for coordinator to load the ingested segments
               Thread.sleep(15000);
        }
}
```

I can probably change the Druid coordinator frequency to run every 1 second to make the loading slightly faster. Right now, it is configured to run every 5 seconds. 
```
druid.coordinator.period=PT1S
```